front/pages/api/v1/w/[wId]/assistant/agent_configuration/[name].ts

```diff
@@ -0,0 +1,116 @@
+import type { WithAPIErrorResponse } from "@dust-tt/types";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { Authenticator, getAPIKey } from "@app/lib/auth";
+import { AgentConfiguration } from "@app/lib/models/assistant/agent";
+import { apiError, withLogging } from "@app/logger/withlogging";
+
+/**
+ * @swagger
+ * /api/v1/w/{wId}/assistant/agent_configuration/{name}:
+ *   get:
+ *     summary: Get a single agent configuration by name
+ *     description: Retrieve a specific agent configuration for the workspace identified by {wId} and {name}.
+ *     tags:
+ *       - Assistant
+ *     parameters:
+ *       - in: path
+ *         name: wId
+ *         required: true
+ *         description: ID of the workspace
+ *         schema:
+ *           type: string
+ *       - in: path
+ *         name: name
+ *         required: true
+ *         description: Name of the agent configuration
+ *         schema:
+ *           type: string
+ *     security:
+ *       - BearerAuth: []
+ *     responses:
+ *       200:
+ *         description: Successfully retrieved agent configuration
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 agentConfiguration:
+ *                   $ref: '#/components/schemas/AgentConfiguration'
+ *       400:
+ *         description: Bad Request. The Assistant API is only available on your own workspace.
+ *       401:
+ *         description: Unauthorized. Invalid or missing authentication token.
+ *       404:
+ *         description: Agent configuration not found.
+ *       405:
+ *         description: Method not supported. Only GET is expected.
+ *       500:
+ *         description: Internal Server Error.
+ */
+
+type GetAgentConfigurationResponseBody = {
+  agentConfiguration: AgentConfiguration | null;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<GetAgentConfigurationResponseBody>>
+): Promise<void> {
+  const keyRes = await getAPIKey(req);
+  if (keyRes.isErr()) {
+    return apiError(req, res, keyRes.error);
+  }
+
+  const { workspaceAuth, keyAuth } = await Authenticator.fromKey(
+    keyRes.value,
+    req.query.wId as string
+  );
+
+  if (
+    !workspaceAuth.isBuilder() ||
+    keyAuth.getNonNullableWorkspace().sId !== req.query.wId
```

Comment:
We should not use the `getNonNullableWorkspace` here as it throws an error. This should mainly be used once you have asserted that there is a valid workspace attached to the user. Which is done automatically on all private api endpoints but not on the public api.

Reply:
Yup, took all that from agent_configurations.ts but I guess is hasn't been touched in a while

Reply:
Done, also edited agent_configurations.ts to reflect this